src/openloco/interop/interop.hpp

```diff
@@ -130,58 +130,179 @@ namespace openloco::interop
     int32_t call(int32_t address);
     int32_t call(int32_t address, registers& registers);
 
-    template<typename T, uint32_t TAddress>
+    template<typename T, uintptr_t TAddress>
     struct loco_global
     {
-        void operator=(T rhs)
+        typedef T type;
+        typedef type* pointer;
+        typedef type& reference;
+        typedef const type& const_reference;
+
+        operator reference()
         {
-            addr<TAddress, T>() = rhs;
+            return addr<TAddress, T>();
         }
 
-        operator T&()
+        loco_global& operator=(const_reference v)
         {
-            return addr<TAddress, T>();
+            addr<TAddress, T>() = v;
+            return *this;
         }
 
-        T& operator*()
+        loco_global& operator+=(const_reference v)
+        {
+            addr<TAddress, T>() += v;
+            return *this;
+        }
+
+        loco_global& operator|=(const_reference v)
+        {
+            addr<TAddress, T>() |= v;
+            return *this;
+        }
+
+        loco_global& operator&=(const_reference v)
+        {
+            addr<TAddress, T>() &= v;
+            return *this;
+        }
+
+        loco_global& operator-=(const_reference v)
+        {
+            addr<TAddress, T>() -= v;
+            return *this;
+        }
+
+        loco_global& operator++()
+        {
+            addr<TAddress, T>()++;
+            return *this;
+        }
+
+        loco_global operator++(int)
+        {
+            addr<TAddress, T>()++;
+            return *this;
+        }
+
+        loco_global& operator--()
+        {
+            addr<TAddress, T>()--;
+            return *this;
+        }
+
+        loco_global operator--(int)
+        {
+            addr<TAddress, T>()--;
+            return *this;
+        }
+
+        reference operator*()
         {
             return addr<TAddress, T>();
         }
 
-        T* operator->()
+        pointer operator->()
         {
             return &(addr<TAddress, T>());
         }
+
+        constexpr size_t size() const
+        {
+            return sizeof(T);
+        }
     };
 
-    template<typename T, size_t TSize, uint32_t TAddress>
-    struct loco_global_array
+    template<typename T>
+    struct loco_global_iterator
     {
-        operator T*()
+    private:
+        T* _ptr;
+
+    public:
+        loco_global_iterator(T* p)
+            : _ptr(p)
+        {
+        }
+        loco_global_iterator& operator++()
+        {
+            ++_ptr;
+            return *this;
+        }
+        loco_global_iterator& operator++(int)
+        {
+            ++_ptr;
```

Comment:
No temporary storage of "before" value for post-increment overload? Looks like a mistake.